### PR TITLE
8353: watch for unevaluated algebraic

### LINKED ISSUE #8353: minpoly should not fail if there is handler for something that _separate_sq fails on
```
>>> e=exp(3*I*pi,evaluate=False);e
exp(3*I*pi)
>>> e.is_algebraic
True
>>> minpoly(e)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\polys\numberfields.py", line 656, in minimal_polynomial
    result = _minpoly_compose(ex, x, dom)
  File "sympy\polys\numberfields.py", line 541, in _minpoly_compose
    ex1 = _separate_sq(ex)
  File "sympy\polys\numberfields.py", line 135, in _separate_sq
    raise NotImplementedError
NotImplementedError  <------------ perhaps catch that error in case there is a handler later ? Not sure.
>>> sympy.polys.numberfields._minpoly_exp(e,x)
x + 1
```

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #8353

#### Release Notes

* polys
  * minpoly no longer errors on unevaluate `exp` numbers (via import `_is_surd_sum`)

### CODE REVIEW

File: sympy/polys/numberfields/tests/test_minpoly.py [hunk 1]
```diff
@@ -186,6 +186,10 @@ def test_issue_26903():
     assert minimal_polynomial(sqrt(2) - zero, x) == x**2 - 2
 
 
+def test_issue_8353():
+    assert minpoly(exp(3*I*pi, evaluate=False), x) == x + 1
```
Comment (smichr) [inline sympy/polys/numberfields/tests/test_minpoly.py:190]:
```suggestion
    assert minimal_polynomial(exp(3*I*pi, evaluate=False), x) == x + 1
```

**Comment by oscarbenjamin** [comment]
I was going to say that I think it looks good but I'm confused about the failing doctest:
```python
In [6]: from sympy.simplify.simplify import nthroot

In [7]: nthroot(90 + 34*sqrt(7), 3)
Out[7]:
3 ____________
╲╱ 34⋅√7 + 90
```
On master it is:
```python
In [3]: nthroot(90 + 34*sqrt(7), 3)
Out[3]: √7 + 3
```
Both answers seem to be equivalent although the result in master is better.

More importantly how does the change here affect this?

Is there an exception being swallowed somewhere in `nthroot` or is `minpoly` returning an incorrect result somewhere?

File: sympy/polys/numberfields/minpoly.py [hunk 2]
```diff
@@ -88,6 +88,15 @@ def _choose_factor(factors, x, v, dom=QQ, prec=200, bound=5):
...
     for y in args:
+        if y.is_Rational:
+            continue
+        # looking for surd or Rational*surd(s)
+        if not y.is_Pow:
+            if not y.is_Mul:
+                return False
+            if not all(i.is_Rational or i.is_Pow for i in y.args):
+                return False
+            return False
```
Comment (smichr) [inline sympy/polys/numberfields/minpoly.py:99]:
```suggestion
```

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**